python/ray/train/_internal/checkpoint.py

```diff
@@ -101,13 +101,20 @@ def _process_checkpoint(
         """Ray Train entrypoint. Perform all processing for a checkpoint."""
         # Get checkpoint from first worker.
         checkpoint_data = checkpoint_results[0].data
+        checkpoint_metadata = checkpoint_results[0].metadata or {}
 
-        # Decode checkpoint.
-        checkpoint_data = decode_checkpoint_fn(checkpoint_data)
+        # TODO(ml-team): Remove once we remove Backend.decode_data
+        checkpoint_data = decode_checkpoint_fn(checkpoint_data).to_dict()
```

Comment:
btw it seems that the semantics for scoring is not consistent across across Tune and our different Trainers.

Currently we do the following:
1. For DL trainers, we check the checkpoint dict itself for score_attribute
2. For non-DL trainers (which goes through `TuneReportCallback`) and for basic Tune, we check the metrics associated with checkpoint for score_attribute, not the checkpoint itself.

With the new API, we should push to using approach 2 for everything in followups.

Reply:
Definitely, let me make a note of that